### PR TITLE
enable diego-cell departed deployment feature to support multi-clusters

### PR DESCRIPTION
**Pull request by SteveZhang-CDL**
## Description
To add the diego-cell separated deployment feature for multi-clusters when it is enabled, only diego-cell will be deployed and it's also compatible for single cluster.

## Motivation and Context
To resolve issue https://github.com/cloudfoundry-incubator/kubecf/issues/934

## How Has This Been Tested?
Enable the diego-cell departed deployment feature and test in IBM IKS clusters.
```
features:
  multiple_cluster_mode:
    multiple_cluster:
      enabled: true
    diego_segment:
      enabled: true
```
The diego-cell is deployed and works fine:

```
ibmadmin@zhlsunshine2:$ helm install kubecf --namespace kubecf --values values-bosh-kubecf.yaml ./
NAME: kubecf
LAST DEPLOYED: Fri Jul  3 20:09:39 2020
NAMESPACE: kubecf
STATUS: deployed
REVISION: 1
TEST SUITE: None
```

```
ibmadmin@zhlsunshine2:$ kubectl get pods -n kubecf
NAME                       READY   STATUS    RESTARTS   AGE
bosh-dns-97764894c-26sc4   1/1     Running   0          5m50s
bosh-dns-97764894c-8d8gt   1/1     Running   0          5m50s
diego-cell-0               7/7     Running   2          5m48s
```

## Screenshots (if appropriate):
I have post 3 screenshots here: https://github.com/cloudfoundry-incubator/kubecf/issues/934#issuecomment-653398769

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code has security implications.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: deploy/helm/kubecf/templates/diego-cell-service.yaml [hunk 1]
```diff
@@ -0,0 +1,81 @@
+{{- if and .Values.features.multiple_cluster_mode.multiple_cluster.enabled .Values.features.multiple_cluster_mode.diego_segment.enabled }}
```
Comment (viovanov) [inline deploy/helm/kubecf/templates/diego-cell-service.yaml:1]: Would `diego-cell-isolation-segment-services.yaml` be a better name for this?

File: deploy/helm/kubecf/templates/diego-cell.yaml [hunk 2]
```diff
@@ -0,0 +1,240 @@
+{{- if and .Values.features.multiple_cluster_mode.multiple_cluster.enabled .Values.features.multiple_cluster_mode.diego_segment.enabled }}
```
Comment (viovanov) [inline deploy/helm/kubecf/templates/diego-cell.yaml:1]: Would `diego-cell-isolation-segment.yaml` be a better name for this?

File: deploy/helm/kubecf/values.yaml [hunk 3]
```diff
@@ -243,6 +243,110 @@ features:
       service:
         # This setting is not currently configurable and must be HIDDEN
         nodePort: 31666
+
+  # To support multi-clusters, the diego-cell would be deployed separately in another different cluster if these two flags, multiple-cluster and diego-segment, are true 
+  multiple_cluster_mode:
+    multiple_cluster:
```
Comment (viovanov) [inline deploy/helm/kubecf/values.yaml:249]: This stutters, can you please remove `multiple_cluster` and rename `multiple_cluster_mode` to `multi_cluster`?
I think the settings should look like:

```
multi_cluster:
  control_plane:
    enabled: false
  cell_segment:
    enabled: true
  ...
```

File: deploy/helm/kubecf/templates/diego-cell-service.yaml (hunk 1 above)
Comment (SteveZhang-CDL) [reply deploy/helm/kubecf/templates/diego-cell-service.yaml:1]: okay, the name will be better

File: deploy/helm/kubecf/templates/diego-cell.yaml (hunk 2 above)
Comment (SteveZhang-CDL) [reply deploy/helm/kubecf/templates/diego-cell.yaml:1]: okay, the name will be better

File: deploy/helm/kubecf/values.yaml (hunk 3 above)
Comment (SteveZhang-CDL) [reply deploy/helm/kubecf/values.yaml:249]: okay, got it.  I will change it.

**Comment by SteveZhang-CDL** [comment]
@viovanov Thank you for your previous review, I have completed the code change and there is new commit, please help to review and merge.

**Comment by viovanov** [comment]
hi @SteveZhang-CDL - can you please fix the linting errors? thanks!

**Comment by SteveZhang-CDL** [comment]
> hi @SteveZhang-CDL - can you please fix the linting errors? thanks!

hi @viovanov ,  I have signed successfully the CLA, and when I click the `lint` Details, I get below: 
image
Maybe need your help on the next process.

**Comment by SteveZhang-CDL** [comment]
Hi @viovanov @fargozhu, For this issue, please let me know if any action should be taken from me. Thanks

**Comment by SteveZhang-CDL** [comment]
Hi @viovanov,   I have fixed the linter error, but there is other linter error which was not made by me. So when I fix it, there is a conflicting file of **deploy/helm/kubecf/values.yaml** ， for now I have resolved the conflicting file, please help to  review it again. Thanks in advanced!

**Comment by jandubois** [comment]
This PR currently fails the `make helmlint` tests.

At the surface-level this is because `features.multiple_cluster.cell_segment.enabled` is `true`.

This doesn't seem to be the correct default, given that the regular chart should not deploy any kind of multi-cluster support. Setting it to `false` fixes (hides) the helm linting errors.

So even after fixing the defaults, you should enable `cell_segment` or `control_plane` individually and run `make helmlint` manually, to make sure the charts are passing.

At a deeper level, there are naming inconsistencies. The templates use `features.multiple_cluster_mode`, but `values.yaml` only defines `features.multiple_clusters`. So it is not quite clear to me how this could work.

**Comment by SteveZhang-CDL** [comment]
@jandubois Thank you very much for your reply,  and I have completed all changes you have mentioned above. BTW, 
@viovanov I check the linter correctly, and deploy diego cell cluster successfully today with the newest version, showing as below snapshots. So please help to check it again. 
Linter

diego-cell

**Pull request merged**